File: src/main/java/fun/bookish/vertx/auth/simple/handler/SimpleAuthHandler.java
package fun.bookish.vertx.auth.simple.handler;


import fun.bookish.vertx.auth.simple.core.SimpleAuthOptions;
import fun.bookish.vertx.auth.simple.provider.SimpleAuthProvider;
import io.vertx.core.Handler;
import io.vertx.core.Vertx;
import io.vertx.ext.web.RoutingContext;

import java.util.Collection;

public interface SimpleAuthHandler extends Handler<RoutingContext> {

    static SimpleAuthHandler create(Vertx vertx, SimpleAuthProvider simpleAuthProvider){
        return new SimpleAuthHandlerImpl(vertx,simpleAuthProvider,new SimpleAuthOptions());
    }

    static SimpleAuthHandler create(Vertx vertx, SimpleAuthProvider simpleAuthProvider, SimpleAuthOptions options){
        return new SimpleAuthHandlerImpl(vertx,simpleAuthProvider,options);
    }

    boolean checkAnno(String permission);

    SimpleAuthHandler addAnnoPermissions(Collection<String> permissions);

    SimpleAuthHandler addAnnoPermission(String permission);

}

File: src/main/kotlin/cn/booklish/vertx/simple/handler/SimpleAuthHandler.kt
package cn.booklish.vertx.simple.handler

import cn.booklish.vertx.simple.provider.SimpleAuthProvider
import io.vertx.core.Handler
import io.vertx.ext.web.RoutingContext

interface SimpleAuthHandler: Handler<RoutingContext> {

    companion object {
        fun create(simpleAuthProvider: SimpleAuthProvider):SimpleAuthHandler{
            return SimpleAuthHandlerImpl(simpleAuthProvider)
        }
    }

    fun checkAnno(permission: String): Boolean

    fun checkCookie(ctx: RoutingContext)

    fun addAnnoPermissions(permissions:List<String>): SimpleAuthHandler

    fun addAnnoPermission(permission:String): SimpleAuthHandler

}